htdocs/fourn/facture/card.php

```diff
@@ -325,7 +325,18 @@
 	// payments conditions
 	if ($action == 'setconditions' && $user->rights->fournisseur->facture->creer)
 	{
-		$result=$object->setPaymentTerms(GETPOST('cond_reglement_id', 'int'));
+	    $object->fetch($id);
+	    $object->cond_reglement_code = 0; // To clean property
+	    $object->cond_reglement_id = 0; // To clean property
+	    $result = $object->setPaymentTerms(GETPOST('cond_reglement_id', 'int'));
```

Comment:
Because this action is composed by 2 different write in database, it must be between a begin and commit transaction.